roles/kubespray-defaults/defaults/main/download.yml

```
@@ -295,7 +295,7 @@ haproxy_image_tag: 2.8.2-alpine
 # Coredns version should be supported by corefile-migration (or at least work with)
 # bundle with kubeadm; if not 'basic' upgrade can sometimes fail
 
-coredns_version: "{{ 'v1.11.1' if (kube_version is version('v1.29.0', '>=')) else 'v1.10.1' }}"
+coredns_version: "{{ 'v1.11.3' if (kube_version is version('v1.29.0', '>=')) else 'v1.10.1' }}"
```

```suggestion
coredns_version: "{{ 'v1.11.3' if (kube_version is version('v1.30.0', '>=')) else 'v1.11.1' }}"
```
Kubernetes v1.29 is on coredns v1.11.1 and 1.[30-31] runs on v1.11.3.